tests/test_containerproperties.py

```diff
@@ -400,3 +419,344 @@ def test_SubElementWithSubElementListProperty(self):
+        self.assertIsNotNone(sub_node)
+        self.assertEqual(sub_node.text, docname_from_qname(new_value, self.node.nsmap))
+
+        # Test removing the sub-element when value is None
+        setattr(instance, self.property._local_var_name, None)
+        self.property.update_xml_value(instance, self.node)
+        sub_node = self.node.find(self.sub_element_name)
+        self.assertIsNone(sub_node)
+
+    def test_update_xml_value_with_none(self):
+        instance = type('TestInstance', (object,), {})()
+        self.property._is_optional = False
+        setattr(instance, self.property._local_var_name, None)
+        with self.assertRaises(ValueError, msg=f'mandatory value {self.sub_element_name.text} missing'):
+            self.property.update_xml_value(instance, self.node)
+
+
+class TestNodeEnumQNameProperty(unittest.TestCase):
+    def setUp(self):
+        self.prefix = utils.random_qname_part()
+        self.q_name = utils.random_qname()
+        self.sub_node_qname = utils.random_qname(localname=self.q_name.localname)
+        self.property = NodeEnumQNameProperty(
+            sub_element_name=self.sub_node_qname,
+            enum_cls=mock.MagicMock(),
+            default_py_value=None,
+            implied_py_value=None,
+            is_optional=True,
+        )
+        self.instance = mock.MagicMock()
+        self.node = etree.Element('Root', nsmap={self.prefix: self.q_name.namespace})
+
+    def test_update_xml_value_with_none(self):
+        setattr(self.instance, self.property._local_var_name, None)
+        self.property.update_xml_value(self.instance, self.node)
+        sub_node = self.node.find(self.sub_node_qname.text, namespaces=self.node.nsmap)
+        self.assertIsNone(sub_node)
+
+    def test_update_xml_value_with_value(self):
+        mock_value = mock.MagicMock()
+        mock_value.value = self.q_name
+        setattr(self.instance, self.property._local_var_name, mock_value)
+        self.property.update_xml_value(self.instance, self.node)
+        sub_node = self.node.find(self.sub_node_qname.text, namespaces=self.node.nsmap)
+        self.assertIsNotNone(sub_node)
+        self.assertEqual(sub_node.text, f'{self.prefix}:{self.q_name.localname}')
+
+    def test_update_xml_value_with_missing_sub_element(self):
+        setattr(self.instance, self.property._local_var_name, None)
+        self.property.update_xml_value(self.instance, self.node)
+        sub_node = self.node.find(self.sub_node_qname.text, namespaces=self.node.nsmap)
+        self.assertIsNone(sub_node)
+
+
+class TestNodeTextQNameListProperty(unittest.TestCase):
+    def setUp(self):
+        self.prefix = utils.random_qname_part()
+        self.sub_element_name = utils.random_qname()
+        self.property = NodeTextQNameListProperty(sub_element_name=self.sub_element_name)
+        self.instance = mock.MagicMock()
+        self.node = etree.Element('Root', nsmap={self.prefix: self.sub_element_name.namespace})

```

Comment:
would suggest to only do:
`self.node = etree.Element('Root')` here and move 
`self.node = etree.Element('Root', nsmap={self.prefix: self.sub_element_name.namespace})` to test `test_update_xml_value_with_values`

Reply:
done